espnet2/train/reporter.py

```diff
@@ -544,18 +544,22 @@ def _plot_stats(self, keys: Sequence[str], key2: str):
 
         return plt
 
-    def tensorboard_add_scalar(self, summary_writer, epoch: int = None):
+    def tensorboard_add_scalar(
+        self, summary_writer, epoch: int = None, key1: str = None
+    ):
         if epoch is None:
             epoch = self.get_epoch()
+            total_count = self.stats[epoch]["train"]["total_count"]
 
-        for key1 in self.get_keys(epoch):
-            for key2 in self.stats[epoch][key1]:
-                if key2 in ("time", "total_count"):
-                    continue
+        if key1 is not None:
+            key1_iterator = tuple([key1])
+        else:
+            key1_iterator = self.get_keys(epoch)
+
+        for key1 in key1_iterator:
+            for key2 in self.get_keys2(key1):
                 summary_writer.add_scalar(
-                    f"{key1}_{key2}_epoch",
-                    self.stats[epoch][key1][key2],
-                    epoch,
+                    f"{key2}", self.stats[epoch][key1][key2], total_count
```

Comment:
I'm not sure but this may conflict with `sub_reporter` (will same key name)?

Reply:
https://github.com/espnet/espnet/blob/c32adcfb2702556c1f16f97098dc89c1e8812b02/espnet2/train/reporter.py#L240

Reply:
We may need to add prefix or suffix.
Or change sub report key naming?

Reply:
> I'm not sure but this may conflict with `sub_reporter` (same key name)?

key_1 is only train or valid. We have now made it a separate reporter. 

> Maybe @kamo-naoyuki designed to `sub_reporter` to monitor value per iter, and `reporter` to monitor value per epoch. I prefer to use iter as x label but we should use different key name for each reporter.

Yeah. This still happens but we can just report it in the same graph, by always mentioning it in terms of "iter". Because of tasks when you are doing adaptation and other things the meaning of epoch is lost quite a bit. 

We could add another plot about epoch vs # iter. What do you think @kan-bayashi @kamo-naoyuki

Reply:
Ok I have done this. Let me know if that's ok. @kan-bayashi @kamo-naoyuki

Reply:
Looks cool. 

> key_1 is only train or valid. We have now made it a separate reporter.

I mean
https://github.com/siddalmia/espnet/blob/3899998dbf8ba5cb48ac7f30c9d7546b29ae7c48/espnet2/train/reporter.py#L240
is called at 
https://github.com/siddalmia/espnet/blob/3899998dbf8ba5cb48ac7f30c9d7546b29ae7c48/espnet2/train/trainer.py#L677
This name key2 is duplicated with 
https://github.com/siddalmia/espnet/blob/3899998dbf8ba5cb48ac7f30c9d7546b29ae7c48/espnet2/train/trainer.py#L338
https://github.com/siddalmia/espnet/blob/3899998dbf8ba5cb48ac7f30c9d7546b29ae7c48/espnet2/train/reporter.py#L564
So I think we need prefix or suffix.

Reply:
@kan-bayashi this is okay. It just means that we would be plotting the train statistics at both the log interval and at also the end of iteration. 

Since we are viewing everything in terms of number of update the plots would not get bad as it would refer to the same (x,y).

Reply:
OK. Now I understood your intension!